File: algorithms/1137.n-th-tribonacci-number/solution.go
func tribonacci(n int) int {
	if n <= 1 {
		return n
	}

	tri := [3]int{0, 0, 1}
	for i := 2; i < n+1; i++ {
		sum := tri[0] + tri[1] + tri[2]

		tri[0] = tri[1]
		tri[1] = tri[2]
		tri[2] = sum
	}

	return tri[2]
}

File: algorithms/1137.n-th-tribonacci-number/solution.rs
impl Solution {
    pub fn tribonacci(n: i32) -> i32 {
        if n <= 1 {
            return n;
        }

        let mut tri = [0, 0, 1];
        for i in 2..n {
            tri.rotate_left(1);
            tri[2] = tri.iter().sum();
        }

        tri.iter().sum()
    }
}
